### PR TITLE
return proper error for inspect and proxy on unhealthy node

### PR DESCRIPTION
**Pull request by vieux**
Signed-off-by: Victor Vieux <victorvieux@gmail.com>

### CODE REVIEW

**Comment by vieux** [comment]
ping @aluzzardi @dongluochen

File: api/utils.go [hunk 1]
```diff
@@ -64,10 +64,14 @@ func newClientAndScheme(tlsConfig *tls.Config) (*http.Client, string) {
 func getContainerFromVars(c *context, vars map[string]string) (string, *cluster.Container, error) {
 	if name, ok := vars["name"]; ok {
 		if container := c.cluster.Container(name); container != nil {
+			if !container.Engine.IsHealthy() {
+				return name, nil, fmt.Errorf("Container running on unhealthy node: %s", name)
```
Comment (dongluochen) [inline api/utils.go]: nit: It doesn't look right to put container name after `unhealthy node:`. How about showing container name and node name? User would be curious on the failing node.

Comment (vieux) [inline api/utils.go]: @dongluochen what about `fmt.Sprintf("Container %s running on unhealthy node %s", name, container.Engine.Name)` ?

File: test/integration/api/logs.bats [hunk 2]
```diff
@@ -27,3 +27,22 @@ function teardown() {
...
+	# Stop node-0
+	docker_host stop ${DOCKER_CONTAINERS[0]}
+
+	retry 5 1 eval "docker_swarm info | grep -q 'Unhealthy'"
+
+
+	# verify
+	run docker_swarm logs test_container
+	[ "$status" -eq 1 ]
+	[[ "${output}" == *"Container running on unhealthy node"* ]]
```
Comment (dongluochen) [inline test/integration/api/logs.bats]: This would fail now.

**Comment by dongluochen** [comment]
LGTM. Thanks @vieux.

**Comment by abronan** [comment]
LGTM

**Pull request merged**